Runtime/Game/Resources/LootLockerConfig.cs

```diff
@@ -1,37 +1,70 @@
 using System;
 using System.IO;
+using System.Linq;
 #if UNITY_EDITOR
+using System.Collections.Generic;
 using UnityEditor;
 using UnityEditor.PackageManager;
 using UnityEditor.PackageManager.Requests;
 using UnityEditor.PackageManager.UI;
+using LootLocker.Admin;
+using System.Text.RegularExpressions;
 #endif
 using UnityEngine;
 
 namespace LootLocker
 {
 
     public class LootLockerConfig : ScriptableObject
+
     {
 
         private static LootLockerConfig settingsInstance;
 
         public virtual string SettingName { get { return "LootLockerConfig"; } }
 
+#if UNITY_EDITOR
+        private void OnEnable()
+        {
+            ProjectSettings.APIKeyEnteredEvent += WriteConfigToDisk;
+            ProjectSettings.DomainKeyEnteredEvent += WriteConfigToDisk;
+        }
+
+        private void OnDisable()
+        {
+            ProjectSettings.APIKeyEnteredEvent -= WriteConfigToDisk;
+            ProjectSettings.DomainKeyEnteredEvent -= WriteConfigToDisk;
+        }
+#endif
         public static LootLockerConfig Get()
         {
             if (settingsInstance != null)
             {
+#if UNITY_EDITOR
+                if (string.IsNullOrEmpty(settingsInstance.apiKey))
+                {
+                    string filePath = "Assets/LootLockerSDK/Resources/Config/Editor/PersistedLootLockerConfig.txt";
```

Comment:
If we're writing to disk, then why not use something like .ini that is a standard for configuration?

Reply:
Either that, or simply write LootLockerConfig as a serizialized json and then we can deserialize the object straight into an instance when reading?